packages/better-auth/src/plugins/bearer/index.ts

```diff
@@ -13,14 +13,76 @@ interface BearerOptions {
 	 * @default false
 	 */
 	requireSignature?: boolean;
+	/**
+	 * Custom cookie name for the temporary bearer token confirmation cookie.
+	 *
+	 * @default "bearer-token-confirmation"
+	 */
+	cookieName?: string;
 }
 
 /**
  * Converts bearer token to session cookie
  */
 export const bearer = (options?: BearerOptions) => {
+	const bearerConfirmationCookieName =
+		options?.cookieName || "bearer-token-confirmation";
 	return {
 		id: "bearer",
+		endpoints: {
+			getBearerToken: createAuthEndpoint(
+				"/get-bearer-token",
+				{
+					method: "GET",
+					metadata: {
+						client: false,
+					},
+					requireHeaders: true,
+				},
+				async (ctx) => {
+					const cookieString = ctx.headers.get("cookie");
+					if (!cookieString) {
+						return ctx.json({
+							success: false,
+						});
+					}
+					const cookies = cookieString
+						.split(";")
+						.map((cookie) => cookie.trim());
```

Comment:
here you manually parse the cookie, but I guess there's `ctx.getCookie` utility to use